src/main/java/itc/transforms/elastix/ElastixSimilarityTransform3D.java

```diff
@@ -28,5 +28,75 @@
  */
 package itc.transforms.elastix;
 
+import static java.lang.Math.sqrt;
+import static net.imglib2.util.LinAlgHelpers.quaternionToR;
+
 public class ElastixSimilarityTransform3D extends ElastixSimilarityTransform {
+
+    private double getNorm( double[] vector )
+    {
+        double norm = 0;
+        for ( int i=0; i<vector.length; i++ ) {
+            norm += vector[i]*vector[i];
+        }
+
+        if (norm > 0)
+        {
+            norm = sqrt(norm);
+        }
+
+        return norm;
+    }
+
+    public double[][] getRotationMatrix()
+    {
+        // transform parameters only gives the first 3 parts (x, y, z) of the unit quaternion - we must calculate
+        // the scalar part (w) from this
+        // based on elastix / itk source code:
+        // https://github.com/SuperElastix/elastix/blob/c3a254562d6801f1a34a2de0ec324c10632b5884/Common/Transforms/itkAdvancedSimilarity3DTransform.hxx#L126
+        // https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/Common/include/itkVersor.hxx#L444
+        // https://itk.org/Doxygen/html/classitk_1_1Versor.html#a72a0cf968370f95dac5bac84a01f93a0
+
+        double[] axis = new double[3];
+        axis[0] = TransformParameters[0];
+        axis[1] = TransformParameters[1];
+        axis[2] = TransformParameters[2];
+        double norm = getNorm( axis );
+
+        double epsilon = 1e-10;
```

Comment:
Can you add maybe a small comment or a ref to explain what is the role of epsilon ? In which case is this important (maybe it's explained in the refs above, but sorry I did not check them)

Reply:
To be honest, I'm not entirely sure why this is necessary. I was just trying to replicate the elastix source code as close as possible e.g. here: https://github.com/SuperElastix/elastix/blob/c3a254562d6801f1a34a2de0ec324c10632b5884/Common/Transforms/itkAdvancedSimilarity3DTransform.hxx#L149
I split up the references in that section to make it clearer which part is based on which reference